src/bin/stratisd.rs

```diff
@@ -109,8 +133,13 @@ fn run() -> StratisResult<()> {
             }
         }
 
-        // Ask the engine to check its pools
-        engine.borrow_mut().check()
+        // Refresh list of dbus fds to poll for every time. This can change as
+        // D-Bus clients come and go.
+        fds.truncate(dbus_fd_start_idx);
+        fds.extend(dbus_conn.watch_fds().iter().map(|w| w.to_pollfd()));
+
+        let r = unsafe { libc::poll(fds.as_mut_ptr(), fds.len() as libc::c_ulong, poll_timeout) };
```

Comment:
Why does whether or not engine has an eventable determine timeout for all interesting file descriptors?

Reply:
Because if the engine doesn't support events then it needs to poll -- periodically have a chance to check itself to see if it needs to do something. (We were doing this before for StratEngine to call `check()`, which would e.g. extend the flex data device if needed, etc..)

Reply:
I should be more specific...most of those fd's have nothing to do with the engine's eventables. Why is their poll-timeout dependent on the existance of the eventable when it wasn't before?

Reply:
I don't follow -- there's just one timeout for all the fds we're polling. The other fds, the D-Bus fds, are already event-driven so if any of them have activity, we return from poll() immediately and handle them. Once we can avoid polling by making the engine check() calls event-driven, then there's no need to "fake" a DM event every 10 seconds, so the timeout serves no purpose any more.